#! /bin/sh /usr/share/dpatch/dpatch-run
## 91_125-utf8.dpatch by  <seneca@localhost.localdomain>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Setting UTF-8 in HTTP headers

@DPATCH@

diff -Naur ledgersmb-1.2.5-new/LedgerSMB/Form.pm ledgersmb-1.2.5-dirty/LedgerSMB/Form.pm
--- ledgersmb-1.2.5-new/LedgerSMB/Form.pm	2007-05-17 11:26:32.000000000 -0400
+++ ledgersmb-1.2.5-dirty/LedgerSMB/Form.pm	2007-05-28 18:02:25.000000000 -0400
@@ -302,22 +302,21 @@
 
     if ( $ENV{GATEWAY_INTERFACE} ) {
 
+        $self->{charset} ||= "utf-8";
         if ( $self->{stylesheet} && ( -f "css/$self->{stylesheet}" ) ) {
             $stylesheet =
 qq|<link rel="stylesheet" href="css/$self->{stylesheet}" type="text/css" title="LedgerSMB stylesheet" />\n|;
         }
 
-        if ( $self->{charset} ) {
-            $charset =
+        $charset =
 qq|<meta http-equiv="content-type" content="text/html; charset=$self->{charset}" />\n|;
-        }
 
         $self->{titlebar} =
           ( $self->{title} )
           ? "$self->{title} - $self->{titlebar}"
           : $self->{titlebar};
 
-        print qq|Content-Type: text/html\n\n
+        print qq|Content-Type: text/html; charset=utf-8\n\n
 <!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" 
 		"http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
 <html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
